app/src/main/java/com/example/tasky/agenda_details/data/repository/AgendaDetailsRemoteRepositoryImpl.kt

```diff
@@ -21,28 +26,30 @@ import okhttp3.RequestBody.Companion.toRequestBody
 import okio.IOException
 
 class AgendaDetailsRemoteRepositoryImpl(
-    @AuthenticatedApi private val api: TaskyApi
+    @AuthenticatedApi private val api: TaskyApi,
+    private val contentResolver: ContentResolver
 ) : AgendaDetailsRemoteRepository {
     override suspend fun createEvent(
         eventDetails: EventDetails,
-        photosByteArray: List<ByteArray>
+        photos: List<Photo.LocalPhoto>
     ): Result<AgendaItem.Event, DataError.Network> {
         return try {
             val gson = Gson()
             val json = gson.toJson(eventDetails)
             val requestBody = json.toRequestBody("application/json; charset=utf-8".toMediaType())
 
-            val photoParts = photosByteArray.mapIndexed { index, byteArray ->
-                val mimeType = when (byteArray.take(3)) {
-                    listOf(0xFF.toByte(), 0xD8.toByte(), 0xFF.toByte()) -> "image/jpeg"
-                    listOf(0x89.toByte(), 0x50.toByte(), 0x4e.toByte()) -> "image/png"
-                    else -> throw Exception("invalid Image format")
-                }
-                val requestFile =
-                    byteArray.toRequestBody(mimeType.toMediaTypeOrNull(), 0, byteArray.size)
+            val photoParts = photos.mapIndexed { index, localPhoto ->
+                val mimeTypeMap = MimeTypeMap.getSingleton()
+                val type = contentResolver.getType(Uri.parse(localPhoto.uri))
+                val extension = mimeTypeMap.getExtensionFromMimeType(type)
```

Comment:
Strictly speaking, you're violating the single responsibility principle here, since your function has multiple responsibilities (or in other words: multiple reasons to change):
1. The event update logic
2. How mime types are parsed

You don't have to be infinitely strict with that, it has to fit to the project's scale, however here I'd say putting this mime type conversion in a separate class would make sense to keep this reusable, since you probably also need this for creating events.

Reply:
yes good idea! to clean up and make function readable